Controllers updated

WaP2000 opened the pull request:
1. Add "Admin can view adoption application"
2. Add "User can apply to adopt"
3. Add "Admin can approve or reject adoption application (no function input so far)"

WaP2000 committed:
Controllers updated

1. Add "Admin can view adoption application"
2. Add "User can apply to adopt"
3. Add "Admin can approve or reject adoption application (no function input so far)"

Lulu54368 commented on src/main/java/com/itproject/petshome/controller/AdminController.java:
```diff
@@ -35,4 +42,17 @@ public PetDTO updatePet(PetDTO petDTO) {
+
+    @Operation(summary = "approve or reject adoption application")
+    @PutMapping("/adoptionApplication")
+    public AdoptionApplicationDTO updateAdoptionApplication() {
```
AdoptionApplicationId(in path) and AdoptionApplicationDTO(request body) should be added

Lulu54368 commented on src/main/java/com/itproject/petshome/controller/DonatorController.java:
```diff
@@ -20,5 +22,12 @@
+    @PostMapping("/")
+//    Require Front end finish the prototype and discuss what input to make
+//    and if we need any services created
+    public DonationDTO addDonation() {
```
request body should be added

Lulu54368 commented on src/main/java/com/itproject/petshome/controller/DonatorController.java:
```diff
@@ -20,5 +22,12 @@
 @RequestMapping("/api/v1/donator")
```
more annotation should be added (bearer, allargsConstructor)

WaP2000 committed:
Controller updated

0. Attempt to resolve issues from last pull request
1. Add volunteer controller
1.1 Volunteer can apply to volunteer
2. Users can view their donation
3. Admin can view volunteer application

Lulu54368 commented on src/main/java/com/itproject/petshome/controller/AdminController.java:
```diff
@@ -45,14 +47,23 @@ public PetDTO deletePet(PetDTO petDTO) {
+
+    @Operation(summary = "view volunteer applications")
+    @GetMapping("/volunteerApplication/{id}")
+    public List<VolunteerApplicationDTO> viewVolunteerApplications(@RequestParam ApplicationStatus status) {
```
should also have path variable here

WaP2000 replied:
Would it be better to return a list of VolunteerApplicationDTO and remove path variable?
Or should it be returning an instance of VolunteerApplicationDTO and add path variable for that specific VolunteerApplicationDTO?

Lulu54368 replied:
I guess for admin, they should get a list of applications as well as the specific application

Lulu54368 commented on src/main/java/com/itproject/petshome/controller/DonatorController.java:
```diff
@@ -19,15 +23,20 @@
-    public DonationDTO addDonation() {
-        return null;
+    @PostMapping("/donate")
+    public DonationDTO addDonation(@RequestBody DonationDTO donationDTO) {
```
userId and DonationInput should be here(coz Donation has Id which client don't know)

Lulu54368 commented on src/main/java/com/itproject/petshome/controller/VolunteerController.java:
```diff
@@ -0,0 +1,31 @@
+@AllArgsConstructor
+@RestController
+@Validated
+@RequestMapping("/api/v1/user")
```
volunteer here instead of user

Lulu54368 commented on src/main/java/com/itproject/petshome/controller/VolunteerController.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Operation(summary = "apply to volunteer")
+    @PostMapping("/applyToVolunteer")
+    public VolunteerApplicationDTO addVolunteerApplication()  throws UserNotFoundException {
```
request body should be here(eg. reasons)

WaP2000 replied:
I thought only an expression of interest is needed to apply to be a volunteer.

If volunteer application requires reasons and other information, then maybe VolunteerApplicationInput class is needed and also VolunteerApplication in model needs to be changed.

Lulu54368 commented on src/main/java/com/itproject/petshome/service/AdminService.java:
```diff
@@ -1,19 +1,29 @@
+    public AdoptionApplicationDTO updateAdoptionApplication
+            (AdoptionApplicationDTO adoptionApplicationDTO, Long adoptionApplicationId) {
         return null;
     }
```
I think admin can only update the status of AdoptionApplication

WaP2000 replied:
I add a class "UpdateAdoptionApplicationInput" to fix it

WaP2000 committed:
Controller updated

1. Path variable added for "viewVolunteerApplication" in adminController
2. userId and DonationInput are added for "addDonation" in DonatorController
3. RequestMapping path chnages to volunteer in VolunteerController
4. Requestbody added for "addVolunteerApplication" in VolunteerController
5. AdminService uses a new class for its input to ensure that only status of an application can be changed.
